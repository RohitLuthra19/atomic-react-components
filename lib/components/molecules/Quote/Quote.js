// @flow
import React from 'react';
import type { Node } from 'react';
import styled from 'styled-components';
import classnames from 'classnames';
import styles from './Quote.style';

type Props = {
  className: string,
  content: string,
  contentSrc: string,
  author?: string,
  company?: string,
};

const Quote = ({ content, contentSrc, author, company, className, ...otherProps }: Props): Node => {
  return (
    <figure {...otherProps} className={classnames('quote-wrap', className)}>
      <span className="quote__mark">&rdquo;</span>
      <blockquote className="quote__content" cite={contentSrc}>
        {content}
      </blockquote>
      <small className="quote__author">
        {typeof author !== 'undefined' ? <cite>{author}</cite> : ''}
        {typeof author !== 'undefined' && typeof company !== 'undefined' ? <span>, </span> : ''}
        {typeof company !== 'undefined' ? <cite>{company}</cite> : ''}
      </small>
    </figure>
  );
};

Quote.defaultProps = {
  author: undefined,
  company: undefined,
};

export default styled(Quote)`
  ${styles};
`;
export { Quote as QuoteVanilla };
