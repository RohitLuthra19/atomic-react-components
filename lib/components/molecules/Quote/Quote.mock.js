const defaultQuote = {
  className: 'quote',
  content: `Be patient with yourself. Self-growth is tender; it's holy ground. There's no greater investment. Be patient with yourself. Self-growth is tender; it's holy ground. There's no greater investment.`,
  contentSrc: 'https://blog.hubspot.com/sales/18-motivational-quotes-to-start-your-day-list',
  author: 'Stephen Covey',
  company: 'My company',
  contentStyle: {
    color: 'red',
  },
  authorStyle: { color: 'blue' },
  quoteStyle: {},
};

const noAuthorQuote = {
  className: 'quote',
  content: `Be patient with yourself. Self-growth is tender; it's holy ground. There's no greater investment. Be patient with yourself. Self-growth is tender; it's holy ground. There's no greater investment.`,
  contentSrc: 'https://blog.hubspot.com/sales/18-motivational-quotes-to-start-your-day-list',
  company: 'My company',
  contentStyle: {},
  authorStyle: {},
  quoteStyle: {},
};

const noCompanyQuote = {
  className: 'quote',
  content: `Be patient with yourself. Self-growth is tender; it's holy ground. There's no greater investment. Be patient with yourself. Self-growth is tender; it's holy ground. There's no greater investment.`,
  contentSrc: 'https://blog.hubspot.com/sales/18-motivational-quotes-to-start-your-day-list',
  author: 'Stephen Covey',
  contentStyle: {},
  authorStyle: {},
  quoteStyle: {},
};

const noCompanyNoAuthorQuote = {
  className: 'quote',
  content: `Be patient with yourself. Self-growth is tender; it's holy ground. There's no greater investment. Be patient with yourself. Self-growth is tender; it's holy ground. There's no greater investment.`,
  contentSrc: 'https://blog.hubspot.com/sales/18-motivational-quotes-to-start-your-day-list',
  contentStyle: {},
  authorStyle: {},
  quoteStyle: {},
};


export { defaultQuote, noAuthorQuote, noCompanyQuote, noCompanyNoAuthorQuote };
