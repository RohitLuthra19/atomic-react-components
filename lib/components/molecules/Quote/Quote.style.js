import { css } from 'styled-components';

const styles = css`
  display: flex;
  flex-flow: row wrap;
  .quote {
    &__mark {
      flex: 1 5%;
      ${props => (props.markStyle ? props.markStyle : '')};
    }
    &__content {
      ${props => (props.contentStyle ? props.contentStyle : '')};
      flex: 1 95%;
    }

    &__author {
      ${props => (props.authorStyle ? props.authorStyle : '')};
      flex: 1 100%;
      flex-direction: row-reverse;
    }
    ${props => (props.quoteStyle ? props.quoteStyle : '')};
  }
`;

export default styles;
