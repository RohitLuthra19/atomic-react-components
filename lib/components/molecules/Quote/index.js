// @flow

export { default } from './Quote';
export { QuoteVanilla } from './Quote';
