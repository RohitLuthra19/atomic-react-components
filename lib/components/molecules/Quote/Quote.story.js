// @flow
import React from 'react';
import { storiesOf } from '@storybook/react';

import Quote, { QuoteVanilla } from './Quote';
import { defaultQuote, noAuthorQuote, noCompanyQuote } from './Quote.mock'

// $FlowFixMe eslint-disable-next-line extra-rules/no-commented-out-code
storiesOf('Molecules', module).addWithChapters('Quote', {
  chapters: [
    {
      sections: [
        {
          sectionFn: () => (
            <QuoteVanilla {...defaultQuote}
         
            />
          ),
          options: {
            showSource: true,
            allowSourceToggling: true,
            showPropTables: true,
            allowPropTablesToggling: true,
          },
        },
      ],
    },
    {
      title: 'Quote with all fields',
      sections: [
        {
          title: 'With all fields',
          sectionFn: () => <Quote {...defaultQuote} />,
        },
        
      ],
    },
    {
      title: 'Quote without author',
      sections: [
        {
          title: 'No Author and company present',
          sectionFn: () => <Quote {...noAuthorQuote} />,
        },
        
      ],
    },
    {
      title: 'Quote without company',
      sections: [
        {
          title: 'No Company but author present',
          sectionFn: () => <Quote {...noCompanyQuote} />,
        },
        
      ],
    },
  ],
});
