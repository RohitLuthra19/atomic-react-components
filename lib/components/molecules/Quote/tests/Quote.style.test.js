import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';

import Quote from '../Quote';
import Theme from '../../../../styles/theme';

describe('Quote', () => {
  let quoteComp = '';
  const theme = { ...Theme };

  test('should render quote correctly', () => {
    quoteComp = shallow(<Quote theme={theme} />);
    expect(quoteComp).toMatchSnapshot();
  });
});
