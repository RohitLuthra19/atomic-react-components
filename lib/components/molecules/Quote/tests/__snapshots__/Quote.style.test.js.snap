// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote should render quote correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "Quote-rsrbyr-0",
        "isStatic": false,
        "rules": Array [
          "display:flex;flex-flow:row wrap;.quote{&__mark{flex:1 5%;",
          [Function],
          ";}&__content{",
          [Function],
          ";flex:1 95%;}&__author{",
          [Function],
          ";flex:1 100%;flex-direction:row-reverse;}",
          [Function],
          ";}",
          ";",
        ],
      },
      "displayName": "Quote",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "Quote-rsrbyr-0",
      "target": [Function],
      "toString": [Function],
      "usesTheme": false,
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  theme={
    Object {
      "backgroundColorPrimary": "#fff",
      "backgroundColorSecondary": "#eaf2f6",
      "backgroundColorSuccess": "#2e993a",
      "borderColorBoldest": "#c00",
      "borderColorSecondary": "#d9d9d9",
      "brandPrimaryColor": "#006193",
      "buttonBackgroundColorPrimary": "#e57725",
      "buttonBackgroundColorSecondary": "#016fad",
      "buttonColorPrimary": "#ffffff",
      "buttonColorSecondary": "#ffffff",
      "colorOutline": "#999",
      "containerShadowPrimary": "rgba(0,0,0,.2)",
      "fieldBackgroundColor": "#ffffff",
      "fieldBackgroundDisabledColor": "#f3f3f3",
      "fieldBorderColor": "#bfbfbf",
      "fieldBorderDisabledColor": "#bfdfdf",
      "fieldBorderErrorColor": "#c00",
      "fieldBorderFocusColor": "#006193",
      "fieldTextColor": "#222",
      "fontBold": "700",
      "fontBook": "400",
      "fontFamilyDefault": "Arial, Helvetica, sans-serif",
      "fontFamilyIcon": "",
      "fontFamilyPrimary": "Arial, Helvetica, sans-serif",
      "fontFamilySecondary": "Arial, Helvetica, sans-serif",
      "fontMedium": "500",
      "fontNormal": "normal",
      "fontSemiBold": "600",
      "fontSizeBase": "16px",
      "fontSizeButtonPrimary": "24px",
      "fontSizeButtonSecondary": "16px",
      "fontSizeDefault": "12px",
      "fontSizeHighlight": "18px",
      "letterSpacing": "0px",
      "lineHeightBase": "1.5",
      "linkColorPrimary": "#008CBA",
      "linkColorSecondary": "#016fad",
      "maxWidthPageContent": "80rem",
      "textColorBrandPrimary": "#006193",
      "textColorError": "#c00",
      "textColorPlaceholder": "#dfdfdf",
      "textColorPrimary": "#222",
      "textColorSecondary": "#262626",
      "textColorSecondaryLight": "#666",
      "textColorSuccess": "#2e993a",
      "zIndex": Object {
        "zLoader": 1000,
        "zNativeTickSelectIcon": 50,
        "zOverlay": 900,
        "zSticky": 700,
      },
    }
  }
/>
`;
